lib/gollum/views/template_cascade.rb

```diff
@@ -0,0 +1,35 @@
+module Precious
+  module Views
+    module TemplateCascade
+      def template_priority_path
+        @@template_priority_path
+      end
+
+      def template_priority_path=(path)
+        @@template_priority_path = File.expand_path(path)
+        @template = nil
+      end
+
+      def first_path_available(name)
+        priority = "#{template_priority_path}/#{name}.#{template_extension}"
+        default = "#{template_path}/#{name}.#{template_extension}"
+        File.exists?(priority) ? priority : default
+      end
+
+      # Following methods should track lib/mustache/settings.rb from Mustache project.
```

Comment:
I can find the `#template_file` method in https://github.com/mustache/mustache/blob/master/lib/mustache/settings.rb but not the `#partial` method. Is that in some other file?

Reply:
Asking because without a reference to the Mustache method, what happens in the `#partial` method looks a bit obscure. Good to know what we should be mirroring, exactly!

Reply:
Yeah that's a good point, and complicated a bit by the fact that the Mustache lib itself has two (and a half) of them:

- https://github.com/mustache/mustache/blob/070187/lib/mustache/context.rb#L26
- https://github.com/mustache/mustache/blob/070187/lib/mustache.rb#L177,L198 <--

The second link is the one getting overridden in this PR. This is confusing because in Mustache, the `settings.rb` file still just operates on the main `Mustache` class so how the _code_ is able to access the method is slightly disconnected from where the _file_ lives.

Updated the comments in the PR to point to the relevant methods more directly. 👍